lib/fluent/plugin/filter_key_value_parser.rb

```diff
@@ -57,7 +58,8 @@ def regex_filter(line)
     def extracted_keys(line)
       keys = @use_regex ? regex_filter(line) : delimiter_filter(line)
       filtered_keys = @filtered_keys_list.empty? ? keys : keys.slice(*@filtered_keys_list)
-      @filtered_keys_regex.nil? ? filtered_keys : filtered_keys.merge(keys.select{ |k,v| @filtered_keys_regex.match(k.to_s)})
+      filtered_keys = @filtered_keys_regex.nil? ? filtered_keys : filtered_keys.merge(keys.select{ |k,v| @filtered_keys_regex.match(k.to_s)})
+      @add_prefix.nil? ? filtered_keys : filtered_keys.transform_keys!{ |key| @add_prefix + key }
```

Comment:
you need to cast `key` to string or do the below 

```suggestion
      @add_prefix.nil? ? filtered_keys : filtered_keys.transform_keys!{ |key| "#{@add_prefix}.#{key}" }
```